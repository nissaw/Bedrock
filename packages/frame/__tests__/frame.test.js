import React from "react";
import { create } from "react-test-renderer";

import Frame from "../src";

describe("Frame", () => {
  describe("correct usage", () => {
    test("Frame is not null", () => {
      expect(Frame).toBeTruthy();
    });

    it("renders with ratio", () => {
      const frame = create(
        <Frame ratio={[16, 9]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });

    it("renders custom position", () => {
      const frame = create(
        <Frame ratio={[16, 9]} position="top left">
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );
      expect(frame.toJSON()).toMatchSnapshot();
    });
  });

  describe("incorrect usage", () => {
    beforeEach(() => {
      jest.spyOn(console, "error");
      console.error.mockImplementation(() => undefined);
    });
    afterEach(() => {
      console.error.mockRestore();
    });

    it("renders default position with incorrect value", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={[16, 9]} position={true}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });
    it("renders ratio of 1:1 with error if no ratio provided", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("renders ratio of 1:1 with error if ratio is not an array", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={{ 0: 16, 1: 9 }}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("falls back to 1 with error if array of length <1 provided", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={[16]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("falls back to 1 with error if array of length >2 provided", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={[16, 9, 8, 6]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });

    it("falls back to 1 with error if array of not numbers provided", () => {
      expect(console.error).not.toBeCalled();

      const errorStack = create(
        <Frame ratio={["16", "9"]}>
          <img src="https://picsum.photos/5000" alt="random thing" />
        </Frame>
      );

      expect(console.error).toBeCalled();
      expect(errorStack.toJSON()).toMatchSnapshot();
    });
  });
});
