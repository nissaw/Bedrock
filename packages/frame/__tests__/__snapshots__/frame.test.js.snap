// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame correct usage renders custom position 1`] = `
.c0 {
  --d: 16;
  --n: 9;
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  padding-block-end: calc(var(--n) / var(--d) * 100%);
  position: relative;
}

.c0 > * {
  overflow: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > img,
.c0 > video {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: top left;
}

<div
  className="c0"
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame correct usage renders with ratio 1`] = `
.c0 {
  --d: 16;
  --n: 9;
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  padding-block-end: calc(var(--n) / var(--d) * 100%);
  position: relative;
}

.c0 > * {
  overflow: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > img,
.c0 > video {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame incorrect usage falls back to 1 with error if array of length <1 provided 1`] = `
.c0 {
  --d: 16;
  --n: 1;
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  padding-block-end: calc(var(--n) / var(--d) * 100%);
  position: relative;
}

.c0 > * {
  overflow: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > img,
.c0 > video {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame incorrect usage falls back to 1 with error if array of length >2 provided 1`] = `
.c0 {
  --d: 16;
  --n: 9;
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  padding-block-end: calc(var(--n) / var(--d) * 100%);
  position: relative;
}

.c0 > * {
  overflow: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > img,
.c0 > video {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame incorrect usage falls back to 1 with error if array of not numbers provided 1`] = `
.c0 {
  --d: 1;
  --n: 1;
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  padding-block-end: calc(var(--n) / var(--d) * 100%);
  position: relative;
}

.c0 > * {
  overflow: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > img,
.c0 > video {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame incorrect usage renders default position with incorrect value 1`] = `
.c0 {
  --d: 16;
  --n: 9;
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  padding-block-end: calc(var(--n) / var(--d) * 100%);
  position: relative;
}

.c0 > * {
  overflow: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > img,
.c0 > video {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame incorrect usage renders ratio of 1:1 with error if no ratio provided 1`] = `
.c0 {
  --d: 1;
  --n: 1;
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  padding-block-end: calc(var(--n) / var(--d) * 100%);
  position: relative;
}

.c0 > * {
  overflow: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > img,
.c0 > video {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;

exports[`Frame incorrect usage renders ratio of 1:1 with error if ratio is not an array 1`] = `
.c0 {
  --d: 16;
  --n: 9;
  box-sizing: border-box;
  display: block;
  inline-size: 100%;
  padding-block-end: calc(var(--n) / var(--d) * 100%);
  position: relative;
}

.c0 > * {
  overflow: hidden;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-inset-block-start: 0;
  -ms-intb-rlock-start: 0;
  inset-block-start: 0;
  -webkit-inset-block-end: 0;
  -ms-inlrock-end: 0;
  inset-block-end: 0;
  inset-inline-start: 0;
  inset-inline-end: 0;
  inset-block: 0;
  inset-inline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > img,
.c0 > video {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: 50%;
}

<div
  className="c0"
>
  <img
    alt="random thing"
    src="https://picsum.photos/5000"
  />
</div>
`;
